unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+        assert aggregator_explicit.latest_grid_tree == {}
+
+    # *
+    @staticmethod
+    def test_constructor__latest_grid_tree_flat(aggregator_explicit):
+        assert aggregator_explicit.latest_grid_tree_flat == {}
+
+    # *
+    @staticmethod
+    def test_constructor__area_name_uuid_mapping(aggregator_explicit):
+        assert aggregator_explicit.area_name_uuid_mapping == {}
+
+
+class TestStartWebsocketConnection:
+    """Includes methods which test start_websocket_connection method."""
+
+    @staticmethod
+    def test_start_websocket_connection__dispatcher(aggregator_explicit):
+        with patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver") \
+                as mocked_class:
+            aggregator_explicit.start_websocket_connection()
+            mocked_class.assert_called()
+
+    @staticmethod
+    def test_start_websocket_connection__websocket_creation(aggregator_explicit):
+        websocket_uri_test = f"{test_simulation['websockets_domain_name']}" \
+                             + f"/{test_simulation['uuid']}/aggregator/" \
+                             + f"{test_agg['uuid']}/"
+
+        with patch("gsy_e_sdk.aggregator.WebsocketThread") as mocked_class:
+            aggregator_explicit.aggregator_uuid = test_agg["uuid"]
+            aggregator_explicit.start_websocket_connection()
+            mocked_class.assert_called_with(websocket_uri_test, test_simulation["domain_name"],
+                                            aggregator_explicit.dispatcher)
+
+    @staticmethod
+    def test_start_websocket_connection__websocket_start(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        aggregator_explicit.websocket_thread.start.assert_called_once()
+
+    @staticmethod
+    def test_start_websocket_connection__callback_thread(aggregator_explicit):
+        aggregator_explicit.start_websocket_connection()
+        assert aggregator_explicit.callback_thread == "test_set_callback_thread"
+
+
+class TestGetUuidFromAreaName:
+    """Includes methods which test get_uuid_from_area_name method."""
+
+    @staticmethod
+    def test_get_uuid_from_area_name__not_empty_dict(aggregator_explicit):
+        aggregator_explicit.area_name_uuid_mapping = {"TestArea": ["test_area_uuid"]}
+        area_name = "TestArea"
+
+        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
+              return_value="test_area_uuid")
+        assert aggregator_explicit.get_uuid_from_area_name(area_name) == "test_area_uuid"
+
+    @staticmethod
+    def test_get_uuid_from_area_name__empty_dict(aggregator_explicit):
+        aggregator_explicit.area_name_uuid_mapping = {}
+        area_name = "TestArea"
+
+        patch("gsy_e_sdk.aggregator.get_uuid_from_area_name_in_tree_dict",
+              return_value="test_area_uuid")
+        assert aggregator_explicit.get_uuid_from_area_name(area_name) is None
+
+
+class TestCalculateGridFee:
```

Comment:
The subdevision into this class could be confusion in the future, when we actually test the `GridFeeCalculation` class. But here you only test wether the Aggregator actually calls the `GridFeeCalculation` methods and not test its functionality.